corehq/apps/reports/models.py

```diff
@@ -73,33 +73,96 @@
 
 
 class HQUserType(object):
```

Comment:
I was initially skeptical, but you won me over in the end.  Couple suggestions:

You're storing meta-information about each user type in a dict, this feels like a good place for a namedtuple, since the dict keys are always hard-coded, and you have a series of them which should always have the same set of fields.

Since you're changing `HQUserType.ACTIVE` et al anyways, why not make that the source of truth for all this info?  That is, rather than putting the dicts in a `HqUserTypesList`, put each under the appropriate name.  Then rather than `HQUserType.get_friendly_name(HQUserType.COMMTRACK)`, you could reference `HQUserType.COMMTRACK.friendly_name`

I'd also put the index in there (call it say, `code`, maybe), so it's an explicit property of each user type, rather than something inferred from the ordering of a list.  We unfortunately rely on this magic number a lot, so it should be visible.

You'd probably still want them all in some sort of collection to reference en masse or by code, maybe a set (so no one can mistake order for having semantic meaning) and/or a code: user_type dict:
```python
    ALL = {ACTIVE, DEMO_USER, ... }
    by_code = {user_type.code: user_type for user_type in ALL}
```
You could set this up in any one of a number of ways.